src/NationalInstruments.Analyzers/Style/ChainOfMethodsWithLambdasAnalyzer.cs

```diff
@@ -36,25 +36,53 @@ public override void Initialize(AnalysisContext context)
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
             context.RegisterSyntaxNodeAction(
-                AnalyzeSyntaxNode,
+                AnalyzeExpression,
                 SyntaxKind.ExpressionStatement,
                 SyntaxKind.EqualsValueClause,
                 SyntaxKind.Argument,
                 SyntaxKind.ArrowExpressionClause);
+
+            context.RegisterSyntaxNodeAction(
+                AnalyzeArrayInitializer,
+                SyntaxKind.ArrayInitializerExpression);
         }
 
-        private static void AnalyzeSyntaxNode(SyntaxNodeAnalysisContext context)
+        private static void AnalyzeExpression(SyntaxNodeAnalysisContext context)
         {
             var parentSyntaxNode = context.Node;
 
             // Find the invocation expression i.e
             // a method/delegate call or a property access
             // or a chain of them
             var invocationExpressionSyntax = parentSyntaxNode
-                .DescendantNodes()
+                .DescendantNodes(IsNotArrayInitializerSyntax)
                 .OfType<InvocationExpressionSyntax>()
                 .FirstOrDefault();
 
+            AnalyzeInvocationExpression(invocationExpressionSyntax, context.ReportDiagnostic);
+
+            bool IsNotArrayInitializerSyntax(SyntaxNode syntaxNode) =>
```

Comment:
This is duplicated on line 140; make into a static method

Reply:
Fixed [here](https://github.com/ni/csharp-styleguide/pull/65/commits/ddf7d01411a02d42c5cd752cf2aecc9fee28ed70).